tests/code/test_minor.py

```diff
@@ -132,26 +134,345 @@ def test_getCommunityEngagementByTerm():
+                                'username': 'neillz'}
+        
+        neillzEngagementData2 = {"id": 1,
+                                "matched": False, 
+                                "name": 'Databses',
+                                'term': 2,
+                                "type": 'course',
+                                'username': 'neillz'}
+        
+        setCommunityEngagementForUser('add', khattsEngagementData1, 'ramsayb2')
+        
+        khattsEngagements = IndividualRequirement.select()
+        # get count 
+        khattsEngagements.count() == 1
+        assert khattsEngagements[0].course == Course.get_by_id(2)
+        assert khattsEngagements[0].program == None
+
+        # add 4 more engagements and make sure the 5th raises the expected exception 
+        setCommunityEngagementForUser('add', khattsEngagementData2, 'ramsayb2')
+        setCommunityEngagementForUser('add', khattsEngagementData3, 'ramsayb2')
+        setCommunityEngagementForUser('add', khattsEngagementData4, 'ramsayb2')
+
+        with pytest.raises(DoesNotExist):
+            setCommunityEngagementForUser('add', khattsEngagementData5, 'ramsayb2')
+
+
+        # add records for another student and make sure it is added correctly. 
+        setCommunityEngagementForUser('add', neillzEngagementData1, 'ramsayb2')
+        neillzEngagements = IndividualRequirement.select()
+        assert neillzEngagements.count() == 5
+        assert neillzEngagements[4].username_id == 'neillz'
+
+        # add a second record for that other student.
+        setCommunityEngagementForUser('add', neillzEngagementData2, 'ramsayb2')
+        neillz2Engagements = IndividualRequirement.select()
+        assert neillz2Engagements.count() == 6
+        assert neillz2Engagements[3].username_id == 'khatts'
+        assert neillz2Engagements[4].username_id == 'neillz'
+        assert neillz2Engagements[4].course == None
+        assert neillz2Engagements[5].username_id == 'neillz'
+        assert neillz2Engagements[5].course == Course.get_by_id(1)
+
+        # Removing requirement
+        # TODO remove nonexisting
+        setCommunityEngagementForUser('remove', khattsEngagementData1, 'ramsayb2')
+        khattsUpdatedEngagements = list(IndividualRequirement.select())
+        assert khattsUpdatedEngagements[0].course == None
+        assert khattsUpdatedEngagements[0].program == Program.get_by_id(9)
+        
+        transaction.rollback()
+
+
+
+@pytest.mark.integration
+def test_getMinorInterest():
+    with mainDB.atomic() as transaction: 
+        # set every users minor interest to no interest
+        User.update(minorInterest = 0).where(User.minorInterest == 1).execute()
+        noStudentsInterested = getMinorInterest()
+        assert noStudentsInterested == []
+
+        # Add a student who has progress towards the minor. They should not be in returned list
+        User.update(minorInterest = 1).where(User.username == 'khatts').execute()
```

Comment:
Are we just assuming they have progress towards the minor because of test data? If so I think we should probably manually make them have progress towards the minor to be clear instead of being dependent on what is in test data. Unless that doesn't really matter.